### PR TITLE
fix: machine unpair (dev)

### PR DESCRIPTION
**Pull request by ubavic**

### CODE REVIEW

**Comment by ubavic** [comment]
[Machine unpair failing](https://trello.com/c/sy6lzuor/1080-machine-unpair-failing)

File: lib/pairing.js [hunk 1]
```diff
@@ -16,15 +17,23 @@ function pullToken (token) {
...
+      (SELECT name FROM devices WHERE device_id=$2),
+      (SELECT model FROM devices WHERE device_id=$2),
+      (SELECT created FROM devices WHERE device_id=$2),
+      now());
+    DELETE FROM devices WHERE device_id=$2;
+    DELETE FROM machine_pings WHERE device_id=$2;
+    DELETE FROM machine_network_heartbeat WHERE device_id=$2;
+    DELETE FROM machine_network_performance WHERE device_id=$2;`
+
+  return  db.tx(t => t.none(sql, [uuid.v4(), deviceId]))
```
Comment (chaotixkilla) [inline lib/pairing.js]: It's better to keep the individual queries separated, both to simplify versioning control and avoid query clumping, as we already have on some other functions. The pre-commit format works great, apart from not calling `t.batch` on the return value

```suggestion
  const q1 = t.none(`INSERT INTO
    unpaired_devices(id, device_id, name, model, paired, unpaired)
    VALUES (
      $1,
      $2,
      (SELECT name FROM devices WHERE device_id=$2),
      (SELECT model FROM devices WHERE device_id=$2),
      (SELECT created FROM devices WHERE device_id=$2),
      now())`, [uuid.v4(), deviceId])
  const q2 = t.none(`DELETE FROM devices WHERE device_id=$1`, [deviceId])
  const q3 = t.none(`DELETE FROM machine_pings WHERE device_id=$1`, [deviceId])
  const q4 = t.none(`DELETE FROM machine_network_heartbeat WHERE device_id=$1`, [deviceId])
  const q5 = t.none(`DELETE FROM machine_network_performance WHERE device_id=$1`, [deviceId])

  return t.batch([q1, q2, q3, q4, q5])
```

**Review by chaotixkilla — CHANGES_REQUESTED** [review]

File: lib/pairing.js (hunk 1 above)
Comment (ubavic) [reply lib/pairing.js]: Good points. I can't directly accept your suggestion as `t.batch` is doesn't guarantee that DELETE queries will be run after `q1` is completed (that is main reason why I put everything in one query).

File: lib/pairing.js [hunk 2]
```diff
@@ -16,15 +17,27 @@ function pullToken (token) {
...
+        (SELECT model FROM devices WHERE device_id=$2),
+        (SELECT created FROM devices WHERE device_id=$2),
+        now()
+      )`, [uuid.v4(), deviceId])
+    .then(() => {
+        const q1 = t.none(`DELETE FROM devices WHERE device_id=$1`, [deviceId])
+        const q2 = t.none(`DELETE FROM machine_pings WHERE device_id=$1`, [deviceId])
+        const q3 = t.none(`DELETE FROM machine_network_heartbeat WHERE device_id=$1`, [deviceId])
+        const q4 = t.none(`DELETE FROM machine_network_performance WHERE device_id=$1`, [deviceId])
+        return  t.batch([q1, q2, q3, q4])
```
Comment (chaotixkilla) [inline lib/pairing.js]: Extra space here

```suggestion
        return t.batch([q1, q2, q3, q4])
```

**Review by chaotixkilla — APPROVED** [review]

File: lib/pairing.js [hunk 3]
```diff
@@ -16,15 +17,27 @@ function pullToken (token) {
...
+  return db.tx(t => 
+    t.none(`INSERT INTO
+      unpaired_devices(id, device_id, name, model, paired, unpaired)
+      VALUES (
+        $1,
+        $2,
+        (SELECT name FROM devices WHERE device_id=$2),
+        (SELECT model FROM devices WHERE device_id=$2),
+        (SELECT created FROM devices WHERE device_id=$2),
+        now()
```
Comment (RafaelTaranto) [inline lib/pairing.js]: Same as the release PR

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

**Pull request merged**